core/models.py

```diff
@@ -254,3 +257,34 @@ class ScheduledEventRunInstance(BaseEntity):
     def __str__(self):
         return self.scheduled_event.name + ' - ' + self.status
 
+
+class FrozenData(BaseEntity):
+    """Stores table names for "frozen" data"""
+    status_choices = (
+        ('p', 'Pending'),
+        ('r', 'Running'),
+        ('c', 'Completed'),
+        ('e', 'Errored'),
+    )
+
+    parent_db_table = models.CharField(max_length=200, null=False)
+    frozen_db_table = models.CharField(max_length=200, null=False)
+    completed = models.CharField(max_length=1, choices=status_choices, default='p')
+    active = models.BooleanField(default=True)
+    comment = models.CharField(max_length=200, null=False)
+
+
+class SavedQueryData(BaseEntity):
+    """Borrows heavily from FrozenData to store query sets """
+    status_choices = (
+        ('p', 'Pending'),
+        ('r', 'Running'),
+        ('c', 'Completed'),
+        ('e', 'Errored'),
+    )
+    saved_query_db_table = models.CharField(max_length=200, null=True)
+    active = models.BooleanField(default=True)
+    operation = models.ForeignKey(Operation, on_delete=models.CASCADE)
+    full_query = models.TextField(null=False)
+    completed = models.CharField(max_length=1, choices=status_choices, default='p')
```

Comment:
This too...

Reply:
Yes please, you can rename them, both. For now they are largely inconsequential and the new name would still make sense. Reason for adding them was in case performance slows down, we can decide to just log the _create requests_ in the DB (to improve user experience and turn around time) and have a cron running in the background to actually finish the job. Was only trying to look into the future.

Reply:
Yeah, always best to look far ahead ... alright, let me rename them. That's pretty much it 👍